std/experimental/allocator/gc_allocator.d

```diff
@@ -80,14 +80,14 @@ struct GCAllocator
     }
 
     /// Ditto
-    @system bool deallocate(void[] b) shared
+    @trusted bool deallocate(void[] b) shared nothrow pure
```

Comment:
e.g.

```
import std.experimental.allocator.gc_allocator: GCAllocator;

T[] makeArray(T)(ref shared GCAllocator alloc, size_t length)
{
    return new T[length];
}

T[] makeArray(T)(ref shared GCAllocator alloc, size_t length, auto ref T init)
{
    T[] arr = makeArray!T(alloc, length);
    foreach (ref el; arr)
    {
        el = init;
    }
    return arr;
}

@safe nothrow unittest
{
    //import std.experimental.allocator: makeArray;
    int[] arr = makeArray!int(GCAllocator.instance, 10);
    int[] arr2 = makeArray!int(GCAllocator.instance, 2, 1);
}
```

(conflicts with `makeArray`, so we would need to put this in `makeArray`)